Sprint 1/Estimation Documents/General Scope of Work Document.md

```diff
@@ -0,0 +1,77 @@
+#General Scope of Work Document
+
+## 1.0 General Goals
+The general goal of this project is to replace the pen-and-paper tables currently used for sleep self-monitoring with a mobile application that can record the user data and send it to the clinicians and researchers.
+This system must be able to:
+* Receive user input from the application of each user anonymously
+* Send the data to a cloud database
+* Allow clinicians to access the data as an excel spreadsheet
+
+## 2.0 Proposed System
+	### 2.1 Overview
+	The system will consist of two components - the application itself and a cloud-based server running an SQL database.
+		### 2.1.1 Application
+		The application will work on both iOS and android devices [Must decide on which versions]. Each user will have an account with no details that could be possibly linked to their identity (Only unique username and password). The application will allow users to input their sleep data daily using the CSD questionnaire. This data will then be sent to the server for storage and processing.
+		### 2.2.1 Server [Zihao, can you add more detail here]
+		The server will be cloud-based and will receive sleep data from each user. Researchers will be able to download this data as an excel spreadsheet for analysis.
+		
+		In order to achieve the interaction between mobile applications and servers, we set up a webservice to accomplish the following purposes: 1) receive requests from the mobile phone and feedback the results (verify login). 2) receive the data transmitted by the mobile phone and store it to the database on the server side. 3) send information to the mobile phone (loading user's history data). Our server is AWS cloud server, mainly used to store user data, web service is implemented by C#, mobile data is sent to the webservice through the form of SOAP.
+	
+	### 2.2 Functional Requirements
+		**Questionnaire**
+		The questionnaire will use the same questions as the Consensus Sleep Diary-M table written using [Which language should be used?]. Submission of invalid or blank data will highlight the error and request it be fixed before proceeding.
+		
+		**User Accounts** [Please comment with a better system if you disagree]
+		The user accounts will be set up using a unique user-chosen username, a user-chosen password and an email address (in case they forget their password) [If this is a feasible addition]. If the username is already taken, they will be prompted to choose another. The account is only used to ensure where their individual data is stored and isn't attached to their identity.
+		Admin accounts will also be implemented and used to request the data be sent to them. [Should this happen on a PC?]
+		
```

Comment:
not so sure what this Admin account is

Reply:
The system is all good. and mention that account is distinguished by a unique userID(primary key)which is easy to maintain in the database

Reply:
Admin account probably means for the researcher being able to access their user data